Add validate script

Deiz opened the pull request:
For use in conjunction with the lists in this repo, or any other list
of URLs.

The utility iterates through URLs, passes them to the validator API and
returns whether they passed or failed. For failures, a link to the
validator is provided by default, to see why the failure occurred.

For interest's sake I ran the script against the list @pdehaye posted on #92, and 145 of the 216 successfully validated. I think the updated scraper in conjunction with (semi-)automatic validation would make it much easier to maintain catalog-list.txt.

rgrp commented:
Incredibly useful!

rgrp commented:
@pdehaye would you like to review this and then we merge?

Deiz commented:
Right now I've got `--quiet` printing only invalid URLs, and I'm wondering if I should replace that with explicit `--valid` and `--invalid` options to avoid needing to do something like this when only interested in valid URLs:
`python scripts/validate.py --quiet catalog-list.txt | sort catalog-list.txt - | uniq -u`

pdehaye commented:
It looks good, but I think the invalid/valid options are more useful. Would be happy to add (possibly in not so near future), but not sure what is the GitHub process to join a pull request.

rgrp commented:
@pdehaye default here would be to merge and then submit new work via PR.

Deiz commented:
Changed the implementation a bit and added `--valid` and `--invalid`, so generating a list of good URLs from catalog-list.txt is now simply:
`python scripts/validate.py --quiet --valid catalog-list.txt >new`

pdehaye commented:
The validator works for me. This requres a catalog list, which is produced by the scraper. The scraper does not work as is, and needs the changes in #92 .

rgrp commented:
@Deiz @pdehaye apologies for delay in merging. Good to get this in.
